### PR TITLE
feat!: `common:module` -> `common:function`

### PR DESCRIPTION
**Pull request by cdata**
This change renames `common:module` to `common:function`, and establishes a new convention for distinguishing a "compiled" module from an "interpreted" one. In this convention, the WIT for a module target defines a `module` world for the "compiled" variant, and a `virtual-module` world for the "interpreted" variant. In our code, this correlates to a rename of `CommonModule` to `CommonFunction`, and `CommonScript` to `CommonFunctionVm`.

### CODE REVIEW

File: typescript/common/data/package.json [hunk 1]
```diff
@@ -28,15 +28,15 @@
...
   "bugs": {
-    "url": "https://github.com/commontoolsinc/labs/issues"
+    "url": "https://github.com/commontoolsinc/system/issues"
   },
-  "homepage": "https://github.com/commontoolsinc/labs#readme",
+  "homepage": "https://github.com/commontoolsinc/system#readme",
   "dependencies": {},
   "devDependencies": {
-    "@bytecodealliance/jco": "^1.4.0",
+    "@bytecodealliance/jco": "^1.3.1",
```
Comment (jsantell) [inline typescript/common/data/package.json:39]: Is this intended to lower the version? Ditto for other package.jsons

File: rust/common-wit/src/lib.rs [hunk 2]
```diff
@@ -7,36 +7,33 @@ use std::{collections::BTreeMap, str::FromStr};
...
-
 /// A target that some candidate source code may express the implementation of
 #[derive(Clone, Copy, Debug, Serialize, Deserialize)]
 pub enum Target {
     /// The most basic target: a Common Module
-    #[serde(rename = "common:module")]
-    CommonModule,
+    #[serde(rename = "common:function/module")]
+    CommonFunction,
     /// Effectively the same as a Common Module, but intepreted w/o a compile step
```
Comment (jsantell) [inline rust/common-wit/src/lib.rs:26]: A few "Common Module" in comments, may need some updates

**Review by jsantell — APPROVED** [review]

File: typescript/common/data/package.json (hunk 1 above)
Comment (cdata) [reply typescript/common/data/package.json:39]: No, this is an accident; I copied these changes over from another branch-in-progress that I started a while back.

File: rust/common-wit/src/lib.rs (hunk 2 above)
Comment (cdata) [reply rust/common-wit/src/lib.rs:26]: In this case it is intended. Module is an umbrella.

**Review by jsantell — APPROVED** [review]

**Pull request merged**